pkg/chartverifier/checks/checks.go

```diff
@@ -232,3 +231,80 @@ func CanBeInstalledWithoutManualPreRequisites(uri string, _ *viper.Viper) (Resul
 func CanBeInstalledWithoutClusterAdminPrivileges(uri string, _ *viper.Viper) (Result, error) {
 	return notImplemented()
 }
+
+func ImagesAreCertified(uri string, _ *viper.Viper) (Result, error) {
+
+	r := NewResult(false, "")
+
+	images, err := getImages(uri)
+
+	if err != nil {
+		r.SetResult(false, fmt.Sprintf("%s : Failed to get images : %v", ImageCertifyFailed, err))
+	} else if len(images) == 0 {
+		r.SetResult(true, NoImagesToCertify)
+	} else {
+		for _, image := range images {
+
+			registries, repository, version := getImageParts(image)
+
+			if len(registries) == 0 {
+				registries, err = getImageRegistries(repository)
+			}
+
+			if err != nil {
+				r.AddResult(false, fmt.Sprintf("%s : %s : %v", ImageNotCertified, image, err))
+			} else if len(registries) == 0 {
+				r.AddResult(false, fmt.Sprintf("%s : %s", ImageNotCertified, image))
+			} else {
+				certified := false
+				for _, registry := range registries {
+					found, checkImageErr := checkImageInRegistry(repository, version, registry)
+					if found {
+						err = nil
+						certified = true
+						break
+					} else if err == nil {
+						err = checkImageErr
+					}
+				}
+				if !certified {
+					if err != nil {
+						r.AddResult(false, fmt.Sprintf("%s : %s : %v", ImageNotCertified, image, err))
+					} else {
+						r.AddResult(false, fmt.Sprintf("%s : %s", ImageNotCertified, image))
+					}
+				} else {
+					r.AddResult(false, fmt.Sprintf("%s : %s", ImageCertified, image))
+				}
+			}
+		}
+	}
+
+	return r, nil
```

Comment:
we never return any error here, hence why do not we drop it from the function signature?

Reply:
The method signature is required for checks.